src/indexing.jl

```diff
@@ -48,11 +48,11 @@ end
 
 ## getproperty is equivalent to indexing with a `Symbol`
 
-# @propagate_inbounds Base.getproperty(d::AbstractDictionary, s::Symbol) = d[s]
-# @propagate_inbounds function Base.setproperty!(d::AbstractDictionary, s::Symbol, x)
-#     d[s] = x
-#     return x
-# end
+@propagate_inbounds Base.getproperty(d::AbstractDictionary, s::Symbol) = d[s]
```

Comment:
One could probably restrict the `AbstractDictionary` type to ones containing `Symbol` keys only and return an explicit error otherwise.

Reply:
So something like
```suggestion
@propagate_inbounds Base.getproperty(d::AbstractDictionary{T}, s::T) where {T<:Union{Symbol,Integer,String}} = d[s]
Base.getproperty(d::AbstractDictionary{T}, s) where {T} = error("`getproperty` can only be used on dictionaries with keys of type `Symbol`, `Integer` or `String` (got  $(T))")
```